src/scanners.rs

```diff
@@ -65,10 +65,23 @@ pub fn is_ascii_whitespace(c: u8) -> bool {
     (c >= 0x09 && c <= 0x0d) || c == b' '
 }
 
+pub fn is_ascii_whitespace_or_carrot(c: u8) -> bool {
+    println!("wh/cr {} {}", c, ((c >= 0x09 && c <= 0x0d) || c == b' ' || c == b'^'));
```

Comment:
Since we're no longer going to be nesting levels, do we actually need to check for the next caret? Wouldn't we just ignore?